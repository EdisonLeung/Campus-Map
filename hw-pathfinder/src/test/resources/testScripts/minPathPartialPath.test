# Test pathfinding for node with 2 possible paths but the paths will
# combine. Tests if pathfinding finds the best possible graph based on
# cost

# Create graph
CreateGraph partialPath

# Add nodes
AddNode partialPath a
AddNode partialPath b
AddNode partialPath c

# Add edges
AddEdge partialPath a b 1.0
AddEdge partialPath a b 2.0
AddEdge partialPath b c 3.0

# test pathfinding
FindPath partialPath a c