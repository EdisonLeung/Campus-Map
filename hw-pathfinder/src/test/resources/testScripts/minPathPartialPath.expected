# Test pathfinding for node with 2 possible paths but the paths will
# combine. Tests if pathfinding finds the best possible graph based on
# cost

# Create graph
created graph partialPath

# Add nodes
added node a to partialPath
added node b to partialPath
added node c to partialPath

# Add edges
added edge 1.000 from a to b in partialPath
added edge 2.000 from a to b in partialPath
added edge 3.000 from b to c in partialPath

# test pathfinding
path from a to c:
a to b with weight 1.000
b to c with weight 3.000
total cost: 4.000